src/TracingMiddleware/TracingMiddleware.cs

```diff
@@ -3,88 +3,107 @@
+
+                        var req = context.Features.Get<IHttpRequestFeature>();
+                        if (req != null)
                         {
-                           requestId = result.ToString();   
+                            this.tracer.Trace(requestId, $"Request {nameof(req.Method)}", req.Method);
+                            this.tracer.Trace(requestId, $"Request {nameof(req.Body)}", req.Body);
+                            this.tracer.Trace(requestId, $"Request {nameof(req.Headers)}", req.Headers);
+                            this.tracer.Trace(requestId, $"Request {nameof(req.Path)}", req.Path);
+                            this.tracer.Trace(requestId, $"Request {nameof(req.PathBase)}", req.PathBase);
+                            this.tracer.Trace(requestId, $"Request {nameof(req.Protocol)}", req.Protocol);
+                            this.tracer.Trace(requestId, $"Request {nameof(req.QueryString)}", req.QueryString);
+                            this.tracer.Trace(requestId, $"Request {nameof(req.Scheme)}", req.Scheme);
                         }
-                        else
+
+                        var res = context.Features.Get<IHttpResponseFeature>();
+                        if (res != null)
                         {
-                           requestId = Guid.NewGuid().ToString();   
+                            this.tracer.Trace(requestId, $"Response {nameof(res.Body)}", res.Body);
+                            this.tracer.Trace(requestId, $"Response {nameof(res.HasStarted)}", res.HasStarted);
+                            this.tracer.Trace(requestId, $"Response {nameof(res.Headers)}", res.Headers);
+                            this.tracer.Trace(requestId, $"Response {nameof(res.ReasonPhrase)}", res.ReasonPhrase);
+                            this.tracer.Trace(requestId, $"Response {nameof(res.StatusCode)}", res.StatusCode);
                         }
-                    }
-                    else
-                    {
-                        requestId = Guid.NewGuid().ToString();
-                        env["owin.RequestId"] = requestId;
-                    }
-                    var path = env["owin.RequestPath"];
-                    tracer.Trace(requestId, "Request Start: " + path);
-                    var stopWatch = Stopwatch.StartNew();
-                    try
-                    {
-                        await next(env).ConfigureAwait(false);
-                    }
-                    catch (Exception exception)
-                    {
-                        tracer.Trace(requestId, exception.Message);
-                        throw;
-                    }
-                    finally
-                    {
-                        stopWatch.Stop();
 
-                        if (tracer.Filters.All(filter => filter.Invoke(env)))
+                        var conn = context.Features.Get<IHttpConnectionFeature>();

```

Comment:
Reckon we should pull these blocks out into individual methods so we'd have something like

```
TraceRequest(context);
TraceResponse(context);
TraceConnection(context);
```

Instead of this huge block?